module/window_swallowing.lua

```diff
@@ -11,22 +11,32 @@ local helpers = require(tostring(...):match(".*bling") .. ".helpers")
 local window_swallowing_activated = false
 
 -- you might want to add or remove applications here
-local dont_swallow_classname_list = beautiful.dont_swallow_classname_list
+local parent_filter_list = beautiful.parent_filter_list
     or { "firefox", "Gimp", "Google-chrome" }
-local activate_dont_swallow_filter = beautiful.dont_swallow_filter_activated
+local child_filter_list = beautiful.child_filter_list
+    or { }
+local swallowing_filter = beautiful.swallowing_filter
     or true
 
--- checks if client classname matches with any entry of the dont-swallow-list
-local function check_if_swallow(c)
-    if not activate_dont_swallow_filter then
-        return true
-    end
-    for _, classname in ipairs(dont_swallow_classname_list) do
-        if classname == c.class then
-            return false
+-- check if element exist in table
+local function is_in_table(element, table)
+    local res = false
+    for _, value in pairs(table) do
+        if element:match(value) then
+            res = true
+            break
         end
     end
-    return true
+    return res
+end
+
+-- checks if parent's classname can be swallowed
+local function check_swallow(class, list)
+    if not swallowing_filter then
+        return true
+    else
+        return not is_in_table(class, list)
+    end
 end
```

Comment:
nit :

`if X return true else return Y` -> `return X or Y`

```lua
local function check_swallow(class, list)
    return (not swallowing_filter) or (not is_in_table(class, list))
end
```

Lua stops testing value as soon as it found a trusty value. It means that whenever (not swallowing_filter) is true, (not is_in_table(class, list)) will not be executed.

It can be tested with something like this :

```
function Y() print('hello') return true end 

-- X or Y
true or Y() -- true
false or Y() -- hello true
```

Reply:
i'm trying to do this but can't get it to work

```lua
    if
        -- will search for "(parent_client.pid)" inside the parent_pid string
        ( tostring(parent_pid):find("("..tostring(parent_client.pid)..")") )
        and ( (not swallowing_filter)
        and ( (not is_in_table(parent_client.class, parent_filter_list))
        or (not is_in_table(c.class, child_filter_list)) ) )
    then
```

Reply:
making the logic of this has been a headache so i tought of this

swallow filter      { if active check the lists }
parent check func   { returns true if parent isn't in list }
child check func    { returns true if child isn't in list }
swallow parent func { if true will swallow window }

| x              | a            | b           | out            |
|----------------|--------------|-------------|----------------|
| swallow filter | parent check | child check | swallow parent |
| F              | F            | F           | T              |
| F              | F            | T           | T              |
| F              | T            | F           | T              |
| F              | T            | T           | T              |
| T              | F            | F           | F              |
| T              | F            | T           | F              |
| T              | T            | F           | F              |
| T              | T            | T           | T              |

the case where the swallow filter is active and both the child and parent are in the lists should not even happen to begin with...

so to shorten this thing so far what i'm thinking is to have 1 function (or perhaps adding another if statement) that checks the var for the filter and if the filter is on then it bothers checking the classes against the corresponding list, so the function would recieve 2 inputs which will be the classes of the parent and children, at least that is the best idea i had.

Reply:
figured something out but is far from the elegance of a chain of pure and,or, not operators and parens.

Reply:
Thanks for the truth table. Yeah, you're probably right to make a separated function. It makes the code more readable.